src/PAModel/PAConvert/Yaml/YamlWriter.cs

```diff
@@ -45,27 +45,39 @@ public void WriteEndObject()
             }
         }
 
-        public void WriteProperty(string propertyName, bool value)
+        public void WriteProperty(string propertyName, bool value, bool includeEquals = true)
```

Comment:
Should we make this a single function with a templated type (for the value parameter), instead of three identical functions with one different overloaded parameter type?

Reply:
We could, and I think it is an interesting idea. That said, I think the intent of this class is to be explicit where possible about what we can serialize correctly. In this case `int`, `double`, `bool`, and `string`. Using a templated type would imply to a consumer that we can handle any type, but that isn't really the case. We have specific logic in the case of `string` and `bool`, and it could be in the future that we have specific logic for `int` and `double` as well, or any new types which are added. For now, I'd prefer to keep these separate and explicit.

Reply:
Do we need separate functions for nullable types? Can the functions defined for non-nullable primitive types handle nullable types? like the one for int can handle int?

Reply:
The `int` method cannot handle `int?` types. Within this repo, no, there isn't a current need for nullable types. If we need it for something else we can handle that separately or handle the `null` case in the code calling YamlWriter.